schema/module-fields.json

```diff
@@ -0,0 +1,28 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "http://example.com/example.json",
+  "type": "array",
+  "title": "The root schema",
+  "description": "The root schema comprises the entire JSON document.",
+  "default": [],
+  "examples": [
+      []
+  ],
+  "items": [],
+  "additionalItems":
+    {
+      "type": "object",
+      "required": ["type"],
+      "properties": { "type": { "enum": ["boolean", "choice", "color", "cta", "datetime", "email", "date"] } },
+      "allOf": [
+         {"if":{"properties":{"type":{"const":"boolean"}}}, "then":{"$ref": "./boolean.json"}},
+         {"if":{"properties":{"type":{"const":"choice"}}}, "then":{"$ref": "./choice.json"}},
+         {"if":{"properties":{"type":{"const":"color"}}}, "then":{"$ref": "./color.json"}},
+         {"if":{"properties":{"type":{"const":"cta"}}}, "then":{"$ref": "./date.json"}},
+         {"if":{"properties":{"type":{"const":"datetime"}}}, "then":{"$ref": "./datetime.json"}},
+         {"if":{"properties":{"type":{"const":"email"}}}, "then":{"$ref": "./email.json"}},
+         {"if":{"properties":{"type":{"const":"date"}}}, "then":{"$ref": "./date.json"}}
```

Comment:
Relatively simple to keep fields separate and compose them so we could also have a similar setup for theme fields.